### ISSUE TITLE
Bug: false positive "ESLint: Parsing error: Unexpected token <"

### ISSUE THREAD

**Issue by pubmikeb** [eslint/eslint#15056]
### Environment

Node version: 16.9.1
npm version: 7.23.0
Local ESLint version: 
Global ESLint version: 8.0.0 beta 2
Operating System: Windows 10 x64 21H1

### What parser are you using?

Other

### What did you do?

```markdown
Upgraded to ESLint 8.0.0 beta 2
```

### What did you expect to happen?

```markdown
No warning at HTML5 page, like it used to be in ESLint 8.0.0 beta 1


<!DOCTYPE html>
<html lang = "en">
```

### What actually happened?

```markdown
I get «ESLint: Parsing error: Unexpected token <» for


<!DOCTYPE html>
<html lang = "en">
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
Can you please include your configuration file and sample code to reproduce?

ESLint doesn’t parse HTML by default, so this is likely a problem with a parser or plugin.

**Comment by pubmikeb** [comment]
My configuration file:

```json
{
	"env": {
		"browser": true,
		"es2021": true,
		"node": true,
		"worker": true
	},
	"parserOptions": {
		"ecmaFeatures": {
			"globalReturn": true,
			"impliedStrict": true,
			"jsx": true
		},
		"ecmaVersion": "latest",
		"sourceType": "module"
	},
	"rules": {
		"accessor-pairs": "error",
		"array-bracket-newline": [
			"error",
			"consistent"
		],
		"array-bracket-spacing": "error",
		"array-callback-return": "error",
		"arrow-body-style": "error",
		"arrow-parens": "error",
		"arrow-spacing": "error",
		"block-scoped-var": "error",
		"block-spacing": "error",
		"brace-style": "error",
		"camelcase": "error",
		"class-methods-use-this": "error",
		"comma-dangle": "error",
		"comma-spacing": "error",
		"comma-style": "error",
		"complexity": "error",
		"computed-property-spacing": "error",
		"consistent-return": "error",
		"consistent-this": "error",
		"constructor-super": "error",
		"curly": "error",
		"default-case-last": "error",
		"default-param-last": "error",
		"dot-location": [
			"error",
			"property"
		],
		"dot-notation": "error",
		"eol-last": "error",
		"eqeqeq": "error",
		"for-direction": "error",
		"func-call-spacing": "error",
		"func-name-matching": "error",
		"func-names": "error",
		"function-paren-newline": [
			"error",
			"never"
		],
		"func-style": "error",
		"generator-star-spacing": "error",
		"getter-return": "error",
		"grouped-accessor-pairs": "error",
		"guard-for-in": "error",
		"id-denylist": "error",
		"id-length": "error",
		"id-match": "error",
		"implicit-arrow-linebreak": "error",
		"jsx-quotes": "error",
		"key-spacing": "error",
		"keyword-spacing": "error",
		"line-comment-position": "error",
		"linebreak-style": [
			"error",
			"windows"
		],
		"lines-around-comment": "error",
		"lines-between-class-members": "error",
		"max-classes-per-file": "error",
		"max-depth": "error",
		"max-nested-callbacks": "error",
		"max-params": [
			"error",
			5
		],
		"max-statements": [
			"error",
			25
		],
		"max-statements-per-line": "error",
		"multiline-comment-style": "error",
		"multiline-ternary": "error",
		"new-cap": "error",
		"new-parens": "error",
		"newline-per-chained-call": "error",
		"no-alert": "error",
		"no-array-constructor": "error",
		"no-async-promise-executor": "error",
		"no-await-in-loop": "error",
		"no-bitwise": "error",
		"no-caller": "error",
		"no-case-declarations": "error",
		"no-class-assign": "error",
		"no-compare-neg-zero": "error",
		"no-cond-assign": "error",
		"no-confusing-arrow": "error",
		"no-console": "error",
		"no-const-assign": "error",
		"no-constant-condition": "error",
		"no-constructor-return": "error",
		"no-continue": "error",
		"no-control-regex": "error",
		"no-debugger": "error",
		"no-delete-var": "error",
		"no-div-regex": "error",
		"no-dupe-args": "error",
		"no-dupe-class-members": "error",
		"no-dupe-else-if": "error",
		"no-dupe-keys": "error",
		"no-duplicate-case": "error",
		"no-duplicate-imports": "error",
		"no-else-return": "error",
		"no-empty": "error",
		"no-empty-character-class": "error",
		"no-empty-function": "error",
		"no-empty-pattern": "error",
		"no-eq-null": "error",
		"no-eval": "error",
		"no-ex-assign": "error",
		"no-extend-native": "error",
		"no-extra-bind": "error",
		"no-extra-boolean-cast": "error",
		"no-extra-label": "error",
		"no-extra-semi": "error",
		"no-fallthrough": "error",
		"no-floating-decimal": "error",
		"no-func-assign": "error",
		"no-global-assign": "error",
		"no-implicit-coercion": "error",
		"no-implicit-globals": "error",
		"no-implied-eval": "error",
		"no-import-assign": "error",
		"no-inline-comments": "error",
		"no-inner-declarations": "error",
		"no-invalid-regexp": "error",
		"no-invalid-this": "error",
		"no-irregular-whitespace": "error",
		"no-iterator": "error",
		"no-label-var": "error",
		"no-labels": "error",
		"no-lone-blocks": "error",
		"no-lonely-if": "error",
		"no-loop-func": "error",
		"no-loss-of-precision": "error",
		"no-magic-numbers": [
			"error",
			{
				"ignore": [
					0,
					1,
					2,
					3,
					4,
					5,
					6,
					7,
					8,
					9,
					10,
					100,
					1000
				]
			}
		],
		"no-misleading-character-class": "error",
		"no-mixed-operators": "error",
		"no-multi-assign": "error",
		"no-multi-spaces": "error",
		"no-multi-str": "error",
		"no-multiple-empty-lines": "error",
		"no-negated-condition": "error",
		"no-nested-ternary": "error",
		"no-new": "error",
		"no-new-func": "error",
		"no-new-object": "error",
		"no-new-symbol": "error",
		"no-new-wrappers": "error",
		"no-nonoctal-decimal-escape": "error",
		"no-obj-calls": "error",
		"no-octal": "error",
		"no-octal-escape": "error",
		"no-param-reassign": "error",
		"no-promise-executor-return": "error",
		"no-proto": "error",
		"no-prototype-builtins": "error",
		"no-redeclare": "error",
		"no-regex-spaces": "error",
		"no-restricted-exports": "error",
		"no-restricted-globals": "error",
		"no-restricted-imports": "error",
		"no-restricted-properties": "error",
		"no-restricted-syntax": "error",
		"no-return-assign": "error",
		"no-return-await": "error",
		"no-script-url": "error",
		"no-self-assign": "error",
		"no-self-compare": "error",
		"no-sequences": "error",
		"no-setter-return": "error",
		"no-shadow": "error",
		"no-shadow-restricted-names": "error",
		"no-sparse-arrays": "error",
		"no-template-curly-in-string": "error",
		"no-this-before-super": "error",
		"no-throw-literal": "error",
		"no-trailing-spaces": "error",
		"no-undef": "error",
		"no-undef-init": "error",
		"no-undefined": "error",
		"no-unexpected-multiline": "error",
		"no-unmodified-loop-condition": "error",
		"no-unneeded-ternary": "error",
		"no-unreachable": "error",
		"no-unreachable-loop": "error",
		"no-unsafe-finally": "error",
		"no-unsafe-negation": "error",
		"no-unsafe-optional-chaining": "error",
		"no-unused-expressions": "error",
		"no-unused-labels": "error",
		"no-unused-vars": "error",
		"no-use-before-define": "error",
		"no-useless-backreference": "error",
		"no-useless-call": "error",
		"no-useless-catch": "error",
		"no-useless-computed-key": "error",
		"no-useless-concat": "error",
		"no-useless-constructor": "error",
		"no-useless-escape": "error",
		"no-useless-rename": "error",
		"no-useless-return": "error",
		"no-var": "error",
		"no-void": "error",
		"no-warning-comments": "error",
		"no-whitespace-before-property": "error",
		"no-with": "error",
		"nonblock-statement-body-position": "error",
		"object-curly-newline": "error",
		"object-curly-spacing": "error",
		"object-property-newline": "error",
		"object-shorthand": "error",
		"one-var-declaration-per-line": "error",
		"operator-assignment": "error",
		"operator-linebreak": [
			"error",
			"before"
		],
		"padded-blocks": "error",
		"padding-line-between-statements": "error",
		"prefer-arrow-callback": "error",
		"prefer-const": "error",
		"prefer-destructuring": "error",
		"prefer-exponentiation-operator": "error",
		"prefer-named-capture-group": "error",
		"prefer-numeric-literals": "error",
		"prefer-object-spread": "error",
		"prefer-promise-reject-errors": "error",
		"prefer-regex-literals": "error",
		"prefer-rest-params": "error",
		"prefer-spread": "error",
		"prefer-template": "error",
		"quote-props": [
			"error",
			"consistent-as-needed"
		],
		"quotes": "error",
		"radix": "error",
		"require-atomic-updates": "error",
		"require-await": "error",
		"require-unicode-regexp": "error",
		"require-yield": "error",
		"rest-spread-spacing": "error",
		"semi": "error",
		"semi-spacing": "error",
		"semi-style": "error",
		"sort-keys": "error",
		"sort-vars": "error",
		"space-before-blocks": "error",
		"space-in-parens": "error",
		"space-infix-ops": "error",
		"space-unary-ops": "error",
		"spaced-comment": "error",
		"strict": "error",
		"switch-colon-spacing": "error",
		"symbol-description": "error",
		"template-curly-spacing": "error",
		"template-tag-spacing": "error",
		"unicode-bom": "error",
		"use-isnan": "error",
		"valid-typeof": "error",
		"vars-on-top": "error",
		"wrap-iife": "error",
		"wrap-regex": "error",
		"yield-star-spacing": "error"
	}
}
```

HTML:

```html
<!DOCTYPE html>
<html lang = "en">

<head>
	<meta content = "default-src 'self' https 'unsafe-eval' 'unsafe-inline'; child-src 'none'; object-src 'none'; report-to policyViolationReport;"
		  http-equiv = "Content-Security-Policy" />
	<meta charset = "utf-8" />
	<title>My App</title>
	<link href = "favicon.ico"
		  rel = "icon"
		  type = "image/x-icon" />
</head>

<body id = "content">
</body>

</html>
```

> _this is likely a problem with a parser or plugin_

How can I check what parser/plugin leads to the issue?

**Comment by mdjermanovic** [comment]
> ```gfm
> I get «ESLint: Parsing error: Unexpected token <» for
> ```

Does this error appear in your IDE or when you run ESLint on the command line?

**Comment by pubmikeb** [comment]
> Does this error appear in your IDE or when you run ESLint on the command line?

In IDE, I'll check if via CLI ESLint returns such error as well.

**Comment by mdjermanovic** [comment]
> In IDE, I'll check if via CLI ESLint returns such error as well.

Thanks!

To clarify - you don't intend to lint HTML files with ESLint, but your IDE is showing ESLint errors when you open a HTML file?

**Comment by pubmikeb** [comment]
> but your IDE is showing ESLint errors when you open a HTML file?

Exactly!

**Comment by mdjermanovic** [comment]
This is likely a problem in the IDE or how it is configured to integrate with ESLint, I don't see anything in v8.0.0-beta.2 changelog that could possibly change this after upgrading from v8.0.0-beta.1.

**Comment by pubmikeb** [comment]
> _when you run ESLint on the command line?_

According to the IDEA team, `eslint test.html` throws the same error.

**Comment by nzakas** [comment]
Converting to a discussion, as this is an IDE issue and not an ESLint issue.
